src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/AvoidUninstantiatedInternalClassesTests.cs

```diff
@@ -1511,6 +1511,87 @@ public void M()
             }.RunAsync();
         }
 
+        [Fact, WorkItem(1878, "https://github.com/dotnet/roslyn-analyzers/issues/1878")]
+        public async Task CA1812_VisualBasic_StaticLikeClass_NoDiagnostic()
+        {
+            await VerifyVB.VerifyAnalyzerAsync(@"
+Imports System
+Friend NotInheritable Class C1
+
+    Private Sub New()
+    End Sub
+
+    Public Shared Function GetSomething(o As Object) As String
+        Return o.ToString()
+    End Function
+End Class
+
+Public Class Helpers
+    Private NotInheritable Class C2
+
+        Private Sub New()
+        End Sub
+
+        Public Shared Function GetSomething(o As Object) As String
+            Return o.ToString()
+        End Function
+    End Class
+End Class
+
+Friend NotInheritable Class C3
+
+    Private Const SomeConstant As String = ""Value""
+    Private Shared f As Integer
+
+    Private Sub New()
+    End Sub
+
+    Public Shared Sub M()
+    End Sub
+
+    Public Shared Property P As Integer
+
+    Public Shared Event ThresholdReached As EventHandler
+End Class
+
+Friend Class C4
+
+    Private Sub New()
+    End Sub
+
+    Public Shared Function GetSomething(o As Object) As String
+        Return o.ToString()
+    End Function
+End Class
+
+Friend NotInheritable Class C5
+
+    Public Sub New()
```

Comment:
The helper is not considering the default constructor visibility and there was no note on if there was a technical motivation to do so, hence I have decided not to change the behavior. We could revisit if we decide we want to have `CA1812` on this type.

Reply:
@sharwell You wrote the helper, was there any technical motivation?